indico/modules/events/registration/controllers/management/__init__.py

```diff
@@ -73,3 +76,43 @@ def _process_args(self):
                               RegistrationFormField.is_enabled,
                               ~RegistrationFormField.is_deleted)
                       .one())
+
+
+# TODO The signals added here are for testing purposes and will be removed
+@signals.event.registration.generate_ticket_qr_code.connect
+def _inject_custom_qr_code_data(registration, person, ticket_data):
+    ticket_data.clear()
+    return {
+        'r': str(registration.id),
+        'f': str(registration.registration_form_id),
+    }
+
+
+QR_CODE_NAME = 'Example'
+
+
+@signals.event.registration.generate_regform_ticket_config_qr_code.connect
+def _inject_pattern_for_ticket_qr_code_config(regform, **kwargs):
+    qr_data = kwargs.get('qr_data', {})
+    qr_data.update({
+        'regex': {
```

Comment:
Since this is structured data, I'd rather have the plugin provide the regex and name, instead of freeform data to update the QR code.